### PR TITLE
Feature/student

### PR DESCRIPTION
**Pull request by kevit-dharmik-rathod**

### CODE REVIEW

File: src/modules/students/student.model.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  name: {
+    type: String,
+    required: true
+  },
+  email: {
+    type: String,
+    required: true,
+    unique: true
+  },
+  role: {
```
Comment (kevit-dev-bhayani) [inline src/modules/students/student.model.ts:20]: give default value for student

Comment (kevit-dharmik-rathod) [reply src/modules/students/student.model.ts:20]: changes according your instruction

File: src/modules/students/student.model.ts [hunk 2]
```diff
@@ -19,7 +19,8 @@ const studentSchema = new Schema({
   },
   role: {
     type: String,
-    required: true
+    required: true,
```
Comment (kevit-dev-bhayani) [inline src/modules/students/student.model.ts]: dont need require

File: src/modules/students/student.controllers.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import {Request, Response, NextFunction} from 'express';
+import bcrypt from 'bcryptjs';
+import {createStudent, findByEmail, findStudentById, findStudents, findAndDelete} from './student.services';
+import {logger} from '../../utils/logger';
+import {utilityError} from '../../utils/utility-error-handler';
+
```
Comment (kevit-dev-bhayani) [inline src/modules/students/student.controllers.ts:6]: write proper code commenting

**Review by kevit-dev-bhayani — CHANGES_REQUESTED** [review]
do this for all

**Review by kevit-dev-bhayani — APPROVED** [review]

**Pull request merged**